### ISSUE TITLE
Selected to clean a project inside cargo's checkout directory

### ISSUE THREAD

**Issue by brson** [dnlmlr/cargo-clean-all#2]
Running `cargo clean-all` on my system I see this suggestion:

```
  wasm_kernel : /home/brian/.cargo/git/checkouts/wasmi-7c64eb2d2cc9c309/7b1f235/benches/wasm/wasm_kernel
      2022-08-29 12:08, 798.53 KB
```

It's the only one inside my `.cargo` directory. Presumably wasmi accidentally packaged a target directory, but I haven't looked closely. It seems like deleting stuff from inside cargo's checkout directory could be disruptive to cargo.

**Comment by brson** [comment]
A command line option to let me keep projects by name, or directory, like `--keep-project wasm_kernel` could help here. I don't want to delete this one but I do want to delete the 70 other suggestions, and I don't see how.

**Comment by brson** [comment]
I just freed 139 GB from my home dir. Awesome project. Thanks.

**Comment by dnlmlr** [comment]
Interesting finding! 
I managed to track down the dependency to [this project](https://github.com/Robbepop/wasm_kernel/tree/6abd5b38d9775b3957087fd9b1e8cae9418602a0/target/wasm32-unknown-unknown/release), which indeed seems to ship a target directory with a precompiled wasm file (I suspect it might actually on purpose).

Regardless I agree that the `~/.cargo` directory should not be affected by a standard cleanup. I think I'll exclude this directory by default when running `cargo clean-all` from a parent directory. This way it can still be cleaned if someone really wants to, by simply going into the diretory before running the command.

An option to ignore a specific project is also a really good idea. This can also be expanded in a way that allows to manually ignore or unignore projects after the initial scan, without the need to redo the whole scan. 
I will add both of these features together with a couple quality of life & UI improvements in a somewhat larger update in the near future. The current version was mainly what I built for myself and it was left in a *good enough to be useful* state, but I plan to add a bit of polish.

Anyways, thanks for the feedback and also for reminding me to continue working on this project.

**Comment by dnlmlr** [comment]
I just published the new version `0.5.0` which now skips over `.cargo` directories and also adds the option to use `--ignore [directory]` in order to ignore a given directory with all subdirectories. There is also a new interactive mode (`-i`) that allows you to manually select and deselect projects after the initial filtering from the commandline.

You can install it as always with `cargo install cargo-clean-all`.

If there are any other problems, feel free to open a new issue

**Comment by brson** [comment]
Thanks @dnlmlr
